apps/src/code-studio/progressRedux.js

```diff
@@ -171,6 +171,20 @@ function bestResultLevelId(levelIds, progressData) {
   return bestId;
 }
 
+/**
+ * Does some processing of our passed in stages, namely
+ * - Removes 'hidden' field
+ * - Adds 'stageNumber' field for non-lockable stages
+ */
+export function processedStages(stages) {
+  let numberOfNonLockableStages = 0;
+
+  return stages.map(stage => ({
+    ..._.omit(stage, 'hidden'),
+    stageNumber: (stage.lockable ? undefined : (++numberOfNonLockableStages))
```

Comment:
Is it bad that I feel a little squicky about the pre-increment in this expression?  I understand how it works, and it's tested, but it just makes me...
![image](https://cloud.githubusercontent.com/assets/1615761/23316435/447c8d40-fa7f-11e6-9190-c8c9088e8985.png)
I'd maybe be less clever and use a block instead of an expression in your map function.  Your call though.